src/main/scala/cromwell/webservice/CallCachingParameters.scala

```diff
@@ -0,0 +1,113 @@
+package cromwell.webservice
+
+import cromwell.engine.WorkflowId
+import cromwell.engine.db.DataAccess._
+import cromwell.engine.db.{DataAccess, ExecutionDatabaseKey}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+import scalaz.Scalaz._
+import scalaz.{Failure, Success, ValidationNel}
+
+case class CallCachingParameters private(workflowId: WorkflowId, callKey: Option[ExecutionDatabaseKey], allow: Boolean)
+
+object CallCachingParameters {
+
+  private [webservice] def validateRecognizedKeys(queryParameters: QueryParameters): ValidationNel[String, Unit] = {
```

Comment:
Now I think about it, I recall liking the `ErrorOr` alias, as it shows up as `ErrorOr[FooBlah]` which is pretty descriptive.